### ISSUE TITLE
Add Flower Baseline: FedMeta

### ISSUE THREAD

**Issue by jinsoogod** [adap/flower#2224]
### Paper

Federated Meta-Learning with Fast Convergence and Efficient Communication

### Link

https://arxiv.org/abs/1802.07876

### Maybe give motivations about why the paper should be implemented as a baseline.

FedMeta was introduced at the above paper and it proposed a foundation framework for the combined field of federated learning and meta-learning research. It combines the two meta-learning approaches, MAML and Meta-SGD, with the principles of federated learning. In subsequent research papers such as Per-FedAvg and Ditto, the methods(FedMeta) introduced in this paper have also been cited. I believe it will serve as the baseline for essential step in personalization through the integration of federated learning and meta-learning in Flower. 

### Is there something else you want to add?

I plan to implement the "FedMeta" algorithm as described in Algorithm 1 of the above paper at  Flower baseline and I make baseline support the the results presented in Table 2 and Figure 2 of the above paper.

### For first time contributors

- [ ]  Read the [`first contribution` doc](https://flower.dev/docs/first-time-contributors.html)
- [ ]  Complete the Flower tutorial
- [ ]  Read the Flower Baselines docs to get an overview:
    - [ ]  [https://flower.dev/docs/using-baselines.html](https://flower.dev/docs/using-baselines.html)
    - [ ]  [https://flower.dev/docs/contributing-baselines.html](https://flower.dev/docs/contributing-baselines.html)

### Prepare - understand the scope

- [ ] Read the paper linked above
- [ ] Create the directory structure in Flower Baselines (just the `__init__.py` files and a `README.md`)
- [ ] Before starting to write code, write down all of the specs of this experiment in a README (dataset, partitioning, model, number of clients, all hyperparameters, …)
- [ ] Open a draft PR

### Implement - make it work

- [ ] Implement some form of dataset loading and partitioning in a separate `dataset.py` (doesn’t have to match the paper exactly)
- [ ] Implement the model in PyTorch
- [ ] Write a test that shows that the model has the number of parameters mentioned in the paper
- [ ] Implement the federated learning setup outlined in the paper, maybe starting with fewer clients
- [ ] Plot accuracy and loss
- [ ] Run it and check if the model starts to converge

### Align - make it converge

- [ ] Implement the exact data partitioning outlined in the paper
- [ ] Use the exact hyperparameters outlined in the paper
- [ ] Make it converge to roughly the same accuracy that the paper states
- [ ] Commit the final hyperparameters and plots
- [ ] Mark the PR as ready

**Comment by jafermarq** [comment]
Hello @jinsoogod, It's great to see you want to include `FedMeta` into Flower. We have implemented a new process for adding baselines which you can find it in the `Contributing a new baseline` section: https://github.com/adap/flower/tree/main/baselines#contributing-a-new-baseline
